syntax = "proto3";

package dcn;

// Public-facing minimal definition
message Particle 
{
    string name = 1;
    string feature_name = 2;
    repeated string composite_names = 3;
    string condition_name = 4;
    repeated int32 condition_args = 5;
}

// Internal full record (used in storage, indexing, etc.)
message ParticleRecord 
{
  Particle particle = 1;
  string owner = 2;
}